src/identity.ts

```diff
@@ -151,6 +150,33 @@ export const getIdentities = async (
   };
 };
 
+/**
+ * Deletes all matching identities.
+ *
+ * @param serviceId The service to search for matching identities within.
+ * @param idsOrTags Either a { userId, tenantId } that can be used to search for a matching identity, or a set of
+ * tags that will be used to search. Pass 'null' to delete all identities for a service.
+ */
+export const deleteIdentities = async (
+  serviceId: string,
+  idsOrTags: IEveryAuthUserTenantSet | IEveryAuthTagSet | null
+): Promise<void> => {
+  if (idsOrTags === undefined || (idsOrTags && Object.keys(idsOrTags).length === 0)) {
+    throw new Error(
+      "The 'idsOrTags' parameter, if not null, must specify at least one identity selection criteria to prevent accidentally deleting all identitites. " +
+        "If you really want to delete all identities, specify 'null' for 'idsOrTags'."
+    );
+  }
+  let options: IEveryAuthSearchOptions | undefined = undefined;
+  do {
+    const identities: IEveryAuthIdentitySearch = await getIdentities(serviceId, idsOrTags || {}, options);
+    for (const identity of identities.items) {
+      await deleteIdentity(serviceId, identity.id);
+    }
```

Comment:
Replace with a Promise.all so it happens async instead of serial.

Reply:
fixed (with pageSize 10)